### ISSUE TITLE
The performance of FINCH on Aggregation

### ISSUE THREAD

**Issue by ShenfeiPei** [ssarfraz/FINCH-Clustering#9]
The code is working fine. But the performance I have got is always 0.96536 in terms of NMI (implemented in sklearn.metrics).
The code I run is as follows:

import numpy as np
import scipy.io as sio
from sklearn.metrics import normalized_mutual_info_score as nmi
from .finch import FINCH

data = sio.loadmat("Agg.mat")
X = data["X"]
y_true = data["Y"]
c_true = len(np.unique(y_true))

Y, num_clu, req_y = FINCH(X, req_clust=c_true, distance='euclidean') # or cosine
acc = nmi(y_true, req_y, average_method="max")
print(acc)

Looking forward to your reply

**Comment by ssarfraz** [comment]
Hi the python distance calculation is a bit sensitive to datatypes. since this 2-D toy data is represented by floats the python euclidean distance output seems to be a bit different than matlab. since here we want to cluster 2-D locations you can also convert the data to int and then do clustering, this results in NMI 98%(as reported in the paper) as opposed to 97% you are getting now in python

c, num_clu, req_y = FINCH(X.astype('int32'), req_clust=c_true, distance='euclidean')
nmi(y_true, req_y)
0.9822517523125824

Hope this helps

**Comment by ShenfeiPei** [comment]
I got it. Thanks for your reply.
